libs/partners/openai/langchain_openai/chat_models/base.py

```diff
@@ -2160,3 +2154,34 @@ class OpenAIRefusalError(Exception):
 
     .. versionadded:: 0.1.21
     """
+
+
+def _create_usage_metadata(oai_token_usage: dict) -> UsageMetadata:
+    input_tokens = oai_token_usage.get("prompt_tokens", 0)
+    output_tokens = oai_token_usage.get("completion_tokens", 0)
+    total_tokens = oai_token_usage.get("total_tokens", input_tokens + output_tokens)
+    input_token_details: dict = {
+        "audio": oai_token_usage.get("prompt_tokens_details", {}).get("audio_tokens"),
+        "cache_read": oai_token_usage.get("prompt_tokens_details", {}).get(
+            "cached_tokens"
+        ),
+    }
+    output_token_details: dict = {
+        "audio": oai_token_usage.get("completion_tokens_details", {}).get(
+            "audio_tokens"
+        ),
+        "reasoning": oai_token_usage.get("completion_tokens_details", {}).get(
+            "reasoning_tokens"
+        ),
+    }
+    return UsageMetadata(
+        input_tokens=input_tokens,
+        output_tokens=output_tokens,
+        total_tokens=total_tokens,
+        input_token_details=InputTokenDetails(
```

Comment:
Do you need InputTokenDetails here? Does it do a 2nd instantiation?

Reply:
for lint, could type ignore